app/models/follow.rb

```diff
@@ -0,0 +1,31 @@
+class Follow < ActiveRecord::Base
+  belongs_to :user
+  #TODO Rock&RoR: Check touch usage on cache system
+  belongs_to :followable, polymorphic: true
+
+  validates :user_id, presence: true
+  validates :followable_id, presence: true
+  validates :followable_type, presence: true
+
+  scope(:by_user_and_followable, lambda do |user, followable|
+    where(user_id: user.id,
+          followable_type: followable.class.to_s,
+          followable_id: followable.id)
+  end)
+
+  def self.followed?(user, followable)
+    return false unless user
+    !! by_user_and_followable(user, followable).try(:first)
+  end
+
+  def self.interests_by(user)
```

Comment:
What do you think about using a symbol proc and { |item| .. } syntax to make function body smaller? It would be something like `user.follows.map { |follow| follow.followable.tags.map(&:name) }.flatten.uniq` (the entire function body)